h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -1729,9 +1729,13 @@ else if (pos > min && pos < maxEx && (i==0 || pos != last)) {
 
     int end=Arrays.binarySearch(sortedSplitPoints, maxEx);
     if (end<0) end=-end-1;
-    assert(end>0 && end<= sortedSplitPoints.length);
-    assert(end>=start);
-    assert(sortedSplitPoints[end-1] < maxEx);
+    String splitPointInfo = "Split points info - split points length: "+sortedSplitPoints.length + " values: ";
```

Comment:
This is not needed by a regular execution. Can we execute it only if the assert actually fails?

Reply:
Yes, we can. :)